### DISCUSSION TITLE
Issue semplici per contributori esterni

### DISCUSSION THREAD

**Discussion by MatteoGheza** [iisgiua/giuaschool#760] [Idee e proposte]
In questo periodo di Hacktoberfest, approfittando di una notifica recente che mi ha ricordato del progetto, mi sono chiesto se ci siano delle attività minori (chores, fix CI, testing, traduzione, etc.) aperte a tutti anche senza eccessiva esperienza della codebase.
Se ci fosse qualcosa di simile, sarei interessato a provare, anche per poter collaborare, seppur in modo molto ridotto, ad un progetto open source di questo tipo.

**Comment by trinko** [comment]
E' un'idea molto interessante, provo a suggerire alcune attività.

Per l'ambito di fix CI, relativamente ai workflow su GitHub:
- rendere più performanti le actions usando una immagine di base più leggera;
- eseguire i test (PHPUnit e Behat) su diverse versioni di PHP: 7.4, 8.0, 8.2;
- eseguire i test (PHPUnit e Behat) su diversi database server: MariaDB, MySql 5.7, MySql 8.0, MySql 8.4 (ho rilevato comportamenti diversi tra MariaDB e MySql su alcuni particolari costrutti SQL)

**Comment by MatteoGheza** [comment]
@trinko rimanendo in tema, ho visto che ultimamente le build stanno fallendo per un errore 500.
Potrebbe servire fare upload del file di log nella GHA per fare troubleshooting?  

Invece, per quanto riguarda la questione delle matrici per test con versioni/DB diversi, pensavi di eseguirle per ogni commit o solo prima delle release/al merge di una PR/ogni x giorni/manualmente?

**Reply by trinko** [reply]
> @trinko rimanendo in tema, ho visto che ultimamente le build stanno fallendo per un errore 500. Potrebbe servire fare upload del file di log nella GHA per fare troubleshooting?

No, non è necessario. Appena ho un po' di tempo cerco di capire il motivo dell'errore.

> Invece, per quanto riguarda la questione delle matrici per test con versioni/DB diversi, pensavi di eseguirle per ogni commit o solo prima delle release/al merge di una PR/ogni x giorni/manualmente?

Pensavo di eseguirle solo alla creazione della release o manualmente, per non appesantire gli altri task.
